backend/test_observer/promotion/promoter.py

```diff
@@ -150,15 +141,14 @@ def run_snap_promoter(session: Session, artefact: Artefact) -> None:
                 )
                 continue
 
-            next_risk = CHANNEL_PROMOTION_MAP[artefact.stage.name]
             if (
-                risk == next_risk != artefact.stage.name.lower()
+                risk != artefact.stage.name.lower()
```

Comment:
is it expected that sometimes an artefact gets promoted skipping one of the stages?

Reply:
we've had situations where celery was down or was there some other issue with deployment during which we missed a snap moving from beta to candidate to stable. But I think it's possible to promote a snap skipping a stage through the snapstore